From a6b008a55ab1f9f3445df48eddd36b1a9c474c39 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Mon, 18 Apr 2022 18:21:03 +0200
Subject: [PATCH] SystemUI: Fix alarm icon

* The icon was initially shown but not again after a reboot
* Reason is, that the user setup isn't completed again
  and therefore the variable mCurrentUserSetup will never
  hold true again
* Fix this by calling onUserSetupChanged once manually to
  get a proper initial state
* Similar code exists in other places in SystemUI as well:
  NavigationBar.java, NetworkControllerImpl.java, ...
  -> Looks like this one was forgotten

Change-Id: I02295a70a82696e7c0c6ade93c96e661140886f0
---
 .../android/systemui/statusbar/phone/PhoneStatusBarPolicy.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
index a60f91991ca9..5e4de7b55a0f 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarPolicy.java
@@ -341,6 +341,9 @@ public void init() {
         mRecordingController.addCallback(this);
 
         mCommandQueue.addCallback(this);
+
+        // Get initial user setup state
+        onUserSetupChanged();
     }
 
     @Override
